### PR TITLE
WIP: Zack/match info

### PR DESCRIPTION
**Pull request by nAmKcAz**
Would need https://github.com/Regression-Games/RegressionGames/pull/30 first

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -20,13 +21,45 @@ function configureBot(bot) {
...
+  matchInfoEmitter.on('player_joined', (matchInfo, username, team) => {
+    console.log(`${username} on team: ${team} has joined the match `)
+    this.matchInfo = matchInfo;
+  })
+
+  matchInfoEmitter.on('player_left', (matchInfo, username, team) => {
+    console.log(`${username} on team: ${team} has left the match `)
+    this.matchInfo = matchInfo;
+  })
+
```
Comment (vontell) [inline index.js:54]: Not needed right now, but we could probably have have a generic version of this for people who don't need to respond to individual events, i.e.

```javascript
let matchInfo = null;
matchInfoEmitter.saveUpdatesTo(matchInfo)
```

Comment (vontell) [reply index.js:54]: (here matchInfoEmitter is some subclass of EventEmitter)

File: index.js [hunk 2]
```diff
@@ -340,44 +376,37 @@ function configureBot(bot) {
...
           console.log("Farmer: DeliveryRun: No player available for delivery.. going back to farming")
           farmingDeliveryRun = false;
-          // not in a promise, no setTimeout
         }
       }
 
       // cut more
       if (!farmingDeliveryRun) {
-        findAndDigBlock(undefined, itemType, false, 50).then( async () => {
+        await findAndDigBlock(undefined, itemType, false, 50).then(async () => {
```
Comment (vontell) [inline index.js:400]: Aren't these the changes from the other PR? Does this rely on your other PR?

File: package.json [hunk 3]
```diff
@@ -15,6 +15,7 @@
     "minecraft-data": "^3.15.0",
     "mineflayer-pathfinder": "^2.3.3",
     "prismarine-viewer": "^1.23.0",
+    "rg-match-info": "../RegressionGames/GameMachine/rg-match-info",
```
Comment (vontell) [inline package.json:18]: We can add this package to NPM. I think we may have to pay for the hosting, and they do have the option for private repos, which may be good in the meantime.

**Review by vontell — APPROVED** [review]

**Comment by RG-nAmKcAz** [comment]
@vontell Bot now working with match info npm module

**Pull request merged**